cpp/include/cuvs/neighbors/cagra.hpp

```diff
@@ -1752,4 +1756,469 @@ void serialize_to_hnswlib(raft::resources const& handle,
  * @}
  */
 
+#ifdef CUVS_BUILD_MG_ALGOS
+
+/// \defgroup mg_cpp_index_build ANN MG index build
+
+/// \ingroup mg_cpp_index_build
+/**
+ * @brief Builds a multi-GPU index
+ *
+ * Usage example:
+ * @code{.cpp}
+ * raft::device_resources_snmg clique;
+ * cuvs::neighbors::mg::index_params<cagra::index_params> index_params;
+ * auto index = cuvs::neighbors::cagra::build(clique, index_params, index_dataset);
+ * @endcode
+ *
+ * @param[in] clique a `raft::device_resources_snmg` object specifying the NCCL clique configuration
+ * @param[in] index_params configure the index building
+ * @param[in] index_dataset a row-major matrix on host [n_rows, dim]
+ *
+ * @return the constructed CAGRA MG index
+ */
+auto build(const raft::device_resources_snmg& clique,
```

Comment:
Please don't couple the standard headers to nccl. This is no good. Instead, please just accept a `raft::device_resources` here but provide a function in `raft::resource::nccl_resource` namespace to introspect the resources for a nccl clique (otherwise fail). The snmg build params should be anough of a difference